### ISSUE TITLE
windows: process_events and file_events

### ISSUE THREAD

**Issue by PoppySeedPlehzr** [facebook/osquery#2567]
We need to get `process_events` and `file_events` ported over to Windows. As a start let's get the subscriber/tables implemented so we can turn back on some of our tests, [as discussed in this thread](https://github.com/facebook/osquery/pull/2563).

Going forward we'll need to get the publishers online and functional.

**Comment by mike-myers-tob** [comment]
We have the `ntfs_journal_events` table now on Windows, but for anything better than that, we'd need to implement a Filesystem Minifilter driver (kernel-mode component).

For monitoring process events, the only way to avoid loading a kernel-mode driver (`PsSetCreateProcessNotifyRoutine` — the most common method used by security products, because it can be used to build real-time detection/blocking) would be to change the system's auditing configuration to enable "Audit Process Creation" and "Audit Process Termination" event logging.

**Comment by hilt86** [comment]
surprised this issue has been stagnant so long considering the importance of this table in the security space!

**Comment by lcostantino** [comment]
I've a POC that add Krabs (ETW) + Process Events for Windows (missing terminate) in case somebody want's to use it for a real implementation.
https://github.com/lcostantino/osquery/pull/1/

**Comment by lcostantino** [comment]
Is this something still wanted?

**Comment by mike-myers-tob** [comment]
> Is this something still wanted?

Yes, although now we have another PR, should we prefer one over the other? https://github.com/osquery/osquery/pull/7821

**Comment by lcostantino** [comment]
@mike-myers-tob  cool, i didn't see that. I believe that PR is better to merge , some refactor i included in the referenced POC may be introduced later, given that i didn't use the kernel provider but the user process provider instead i got other information (like Full image Path), and moved lot of of utilities from Processes table (windows) into a common winutil to acquire the same information for processes without duplication.

**Comment by mike-myers-tob** [comment]
@lcostantino those differences sound like valuable contributions to the other PR. If possible, could you leave the suggestions on that PR's feedback, and we'll see what can be done? The author of that PR might be wiling to do a follow-up PR implementing more complex suggestions or a refactor.

**Comment by directionless** [comment]
https://github.com/osquery/osquery/pull/7821 merged, and whatever else we need, we can open anew
